File: Praktikum 6. Array/Soal 5/PRAK605-2310817310001-AvantioFierzaPatria.c
#include <stdio.h>

void multiply_matrices(int n, int A[n][n], int B[n][n], int result[n][n])
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            result[i][j] = 0;
            for (int k = 0; k < n; k++)
            {
                result[i][j] += A[i][k] * B[k][j];
            }
        }
    }
}

void input_matrix(int n, int matrix[n][n])
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            scanf("%d", &matrix[i][j]);
        }
    }
}

void print_matrix(int n, int matrix[n][n])
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

int main()
{
    int n;

    scanf("%d", &n);

    int A[n][n], B[n][n], result[n][n];

    printf("Input matriks A:\n");
    input_matrix(n, A);

    printf("Input matriks B:\n");
    input_matrix(n, B);

    multiply_matrices(n, A, B, result);

    printf("Hasil perkalian matriks A dan B:\n");
    print_matrix(n, result);

    return 0;
}

File: Praktikum 6. Array/Soal 5/PRAK605-2310817310001-AvantioFierzaPatria.py
def multiply_matrices(matrix_a, matrix_b):
    result = [[0 for _ in range(len(matrix_a))] for _ in range(len(matrix_a))]

    for i in range(len(matrix_a)):
        for j in range(len(matrix_a[0])):
            for k in range(len(matrix_b[0])):
                result[i][k] += matrix_a[i][j] * matrix_b[j][k]

    return result


def input_matrix():
    n = int(input())
    matrix = []

    for _ in range(n):
        row = list(map(int, input().split()))
        matrix.append(row)

    return matrix


def print_matrix(matrix):
    for row in matrix:
        print(" ".join(map(str, row)))


print("Input matriks A:")
matrix_a = input_matrix()

print("Input matriks B:")
matrix_b = input_matrix()

result_matrix = multiply_matrices(matrix_a, matrix_b)

print("Hasil perkalian matriks A dan B:")
print_matrix(result_matrix)
